index.js

```diff
@@ -1,88 +1,131 @@
 // console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 3 *****
 // *** Uncomment the lines below to test
 // console.log("%cQuestion 3", "color: red") 
-
-// console.log(drinkWater(12))
+function drinkWater(currentThirstLevel) {
+  console.log('Man I sure am thirsty')
+  console.log('Ahh that hits the spot')
+  return currentThirstLevel - 1
+}
+ console.log(drinkWater(12))
 // // => "Man I sure am thirsty"
 // // => "Ahh that hits the spot"
 // // => 11
-// console.log("%c----------", "color: red") 
+ console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
 // *** Uncomment the lines below to test
 // console.log("%cQuestion 4", "color: red") 
-
-// console.log(sameSameButDifferent(5, 5)) 
+ function sameSameButDifferent (num, maybeNum){
+
+      if (num === maybeNum) { 
+        return "same same"
+      }
+      else if (num == maybeNum){
+        return "same same (but different)"
+      }
+      else {
+        return "different"
+      }
+ }
+ console.log(sameSameButDifferent(5, 5)) 
 // // => "same same"
 
-// console.log(sameSameButDifferent(123, "123")) 
+ console.log(sameSameButDifferent(123, "123")) 
 // // => "same same (but different)"
 
-// console.log(sameSameButDifferent(5, 7)) 
+ console.log(sameSameButDifferent(5, 7)) 
 // // => "different"
 
-// console.log(sameSameButDifferent(123, "122")) 
+console.log(sameSameButDifferent(123, "122")) 
 // // => "different"
-// console.log("%c----------", "color: red") 
+ console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 5 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 5", "color: red")
-// const student1 = { name: "Duane", grade: 88 }
-// updateGrade(student1, 92)
-// console.log(student1)
+console.log("%cQuestion 5", "color: red")
+  function updateGrade (student, grade){
+    student.grade = grade
+}
+const student1 = { name: "Duane", grade: 88 }
+updateGrade(student1, 92)
+console.log(student1)
 // // => { name: "Duane", grade: 92 }
-// console.log("%c----------", "color: red")
+
+
+
+console.log("%c----------", "color: red")
 
 
 
 // ***** Question 6 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 6", "color: red")
-
-// const users = [ 
-//   { 
-//     name: "Duane", phones: { cell: "555-123-4567", office: "555-456-7890" }
-//   },
-//   { 
-//     name: "Liza", phones: { cell: "555-234-5678", office: "555-567-1234" }
+console.log("%cQuestion 6", "color: red")
+// Question - how would I interpolate cell/office in this code?
+// function printNameAndPhones (array) {
+//   if (typeof array === 'object') {
+//     for (const value in array) {
+//       printNameAndPhones(array[value]);
+//     }
+//   } else {
+//     console.log(array);
 //   }
-// ]
-// printNameAndPhones(users)
+// }
+function printNameAndPhones (array){
+  array.forEach(function(e) {
+    console.log(e.name)
+    console.log(`Cell: ${e.phones.cell}`)
+    console.log(`Office: ${e.phones.office}`)
+  } )
+}
```

Comment:
Great! Be careful about your iterator variable. "e" is not particularly descriptive. I would prefer this to be named something like "userObject" or "userData".